docs/roadmap.md

```diff
@@ -0,0 +1,36 @@
+# Release Roadmap
+
+This document helps providing visibility. Starting from release 2.5, Mixxx is following [a 6-month development cycle](https://github.com/mixxxdj/mixxx/wiki/Release-Strategy). This means that every 6 month, a new minor version is released, with the latest 6-month worth of feature, which may vary depending of the community involvement.
+
+This way of working aim to reduce the time it takes for a contributor to see the feature they've worked on being release, while also allowing the Mixxx Core Team not to spend too much time on new release admin tasks.
+
+## Timeline
+
+> [!TIP]
+> Only up to two previous and next releases are kept in this document to keep it manageable. If you are interested in older releases, you may consult the [changelog](../CHANGELOG.md)
+---
+> [!IMPORTANT]
+> The date provided for the future release are purely indicative. The Mixxx Core Team reserves the right to adjust them in order to help with team availability and delays in tasks.
+>
+> Note that the version is only an indication, and depending of the impact of the features changes, the team may decide to release a major version (3.0)
+
+```mermaid
+    %%{init: { 'logLevel': 'debug', 'theme': 'default' , 'themeVariables': {
+              'cScale0': '#323632', 'cScaleLabel0': '#ffffff', 'cScaleInv0': '#555555',
+              'cScale1': '#139013', 'cScaleLabel1': '#ffffff', 'cScaleInv1': '#555555',
+              'cScale2': '#13901344', 'cScaleLabel2': '#ffffff44', 'cScaleInv2': '#77ff7744',
+              'cScale3': '#32363244', 'cScaleLabel3': '#ffffff44', 'cScaleInv3': '#55555544'
+       } } }%%
+timeline
+    section Previous releases
+        2.2.x: 2.2.0 (2018/12/24): 2.2.1 (2019/04/22): 2.2.2 (2019/08/10): 2.2.3 (2019/11/26): 2.2.4 (2020/05/15): 2.2.4.1 (2021/02/17)
+        2.3.x: Beta (2020/05/12): 2.3.0 (2021/06/28): 2.3.1 (2021/09/29): 2.3.2 (2022/01/30): 2.3.3 (2022/06/21): 2.3.4 (2023/03/03): 2.3.5 (2023/05/10): 2.3.6 (2023/08/15)
```

Comment:
whats the advantage of documenting this here when the same information is already visible in the appropriate github release or tag or milestone?

Reply:
Not sure there is any advantage to be honest, I just thought it would make the timeline slightly easier to read but arguable we could truncate anything before the current release if you think this is just noise.

Reply:
> Maybe also on our web page?

I was thinking exactly that but not sure how the team would receive that document in general at first and also how Mermaid is supported in our web framework.

> while the "As long as it takes" has priority (quality over scheduled dates)

100% makes sense. This is what I tried to capture in the info box:
```
> [!IMPORTANT]
> The date provided for the future release are purely indicative. The Mixxx Core Team reserves the right to adjust them in order to help with team availability and delays in tasks.
```
Can you think of a batter way to emphasise this point?

> Is it possible to relate your diagram to the calendar date?

If you mean [that diagram](https://upload.wikimedia.org/wikipedia/de/timeline/bjcrem7rprzoedh3ra7hitacvz3nn4h.png), I don't think Mermaid as any template for it, but I'll confirm that.

> This one is also interesting:
https://launchpad.net/mixxx

Ah that's a bummer. I started making a branching diagram and decided to abandon the idea as it felt too convoluted. I will restart the idea so we can choose which one we prefer

Reply:
I couldn't find a way to make it look like your first example, but I was able to make something similar to the second. I don't find it very good, mainly due to Mermaid limitation, so I would suggest we keep the first one instead.

Reply:
I don't have a strict preference here. Something is better than nothing. We can all the time improve it.